src/main/java/com/scalar/db/storage/jdbc/JDBCService.java

```diff
@@ -0,0 +1,213 @@
+    ResultSet resultSet = preparedStatement.executeQuery();
+    return new ScannerImpl(selectQuery, connection, resultSet);
+  }
+
+  private void addProjectionsForKeys(Selection selection) {
+    if (selection.getProjections().size() == 0) { // meaning projecting all
+      return;
+    }
+    selection.getPartitionKey().forEach(v -> selection.withProjection(v.getName()));
+    selection
+        .getClusteringKey()
+        .ifPresent(k -> k.forEach(v -> selection.withProjection(v.getName())));
+  }
+
+  public boolean put(
+      Put put, Connection connection, @Nullable String defaultSchema, @Nullable String defaultTable)
+      throws SQLException {
+    Table table = getTable(put, defaultSchema, defaultTable);
+    @Nullable Key clusteringKey = put.getClusteringKey().orElse(null);
+    @Nullable MutationCondition condition = put.getCondition().orElse(null);
+
+    operationChecker.checkPut(
+        table, put.getPartitionKey(), clusteringKey, put.getValues(), condition);
+
+    if (condition == null) {
+      queryBuilder
+          .upsertInto(table)
+          .values(put.getPartitionKey(), clusteringKey, put.getValues())
+          .build()
+          .prepareAndBind(connection)
+          .executeUpdate();
+      return true;
+    } else {
+      return new ConditionalUpdater(
+              connection,
+              queryBuilder,
+              table,
+              put.getPartitionKey(),
+              clusteringKey,
+              condition,
+              put.getValues())
+          .update();
+    }
+  }
+
+  public boolean delete(
+      Delete delete,
+      Connection connection,
+      @Nullable String defaultSchema,
+      @Nullable String defaultTable)
+      throws SQLException {
+    Table table = getTable(delete, defaultSchema, defaultTable);
+    @Nullable Key clusteringKey = delete.getClusteringKey().orElse(null);
+    @Nullable MutationCondition condition = delete.getCondition().orElse(null);
+
+    operationChecker.checkDelete(table, delete.getPartitionKey(), clusteringKey, condition);
+
+    if (condition == null) {
+      queryBuilder
+          .deleteFrom(table)
+          .where(delete.getPartitionKey(), clusteringKey)
+          .build()
+          .prepareAndBind(connection)
+          .executeUpdate();
+      return true;
+    } else {
+      return new ConditionalUpdater(
+              connection,
+              queryBuilder,
+              table,
+              delete.getPartitionKey(),
+              clusteringKey,
+              condition,
+              null)
+          .update();
+    }
+  }
+
+  public boolean mutate(
+      List<? extends Mutation> mutations,
+      Connection connection,
+      @Nullable String defaultSchema,
+      @Nullable String defaultTable)
+      throws SQLException {
+    operationChecker.checkMutate(mutations);
+
+    for (Mutation mutation : mutations) {
+      if (mutation instanceof Put) {
+        if (!put((Put) mutation, connection, defaultSchema, defaultTable)) {
```

Comment:
I think `put` and `delete` methods can use the same method name like `mutate` and use overloading so that this part can be simplified without `instanceof` and casting.